easybuild/tools/environment.py

```diff
@@ -221,3 +222,43 @@ def sanitize_env():
     # unset all $PYTHON* environment variables
     keys_to_unset = [key for key in os.environ if key.startswith('PYTHON')]
     unset_env_vars(keys_to_unset, verbose=False)
+
+
+@contextlib.contextmanager
+def wrap_path_env(prepend=None, append=None, sep=os.pathsep, strict=False):
+    """This function is a context manager that temporarily modifies path-like environment variables.
+    It will prepend and append the values of the given dictionaries to the current environment and restore the
+    original environment when the context is exited.
+
+    A custom separator can be specified for each variable by passing a dictionary of strings with the same keys as
+    prepend and append.
+    If a key is not present in the sep dictionary, os.pathsep will be used unless strict is True, then an error
+    will be raised.
```

Comment:
needs docstrings for arguments

Reply:
Fixed with [a0497db](https://github.com/easybuilders/easybuild-framework/pull/4681/commits/a0497db249af707b2e7a4c225e053babecdfde21)